api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineWithPermissionsTest.java

```diff
@@ -71,80 +67,264 @@
+
+        doReturn(Arrays.asList(pipeline1, pipeline2)).when(mockEntityManager).loadAllWithParents(eq(AclClass.PIPELINE), any(), any());
+        doReturn(Arrays.asList(folder1, folder2)).when(mockEntityManager).loadAllWithParents(eq(AclClass.FOLDER), any(), any());
+
 
-        grantPermissionToFolder(AclPermission.READ.getMask(), testUserSid1.getSid(), folder1.getId());
-        grantPermissionToFolder(AclPermission.READ.getMask(), testUserSid1.getSid(), folder2.getId());
-        grantPermissionsToPipeline(AclPermission.WRITE.getMask(), testUserSid1.getSid(), pipeline1.getId());
-        grantPermissionsToPipeline(AclPermission.READ.getMask(), testUserSid1.getSid(), pipeline2.getId());
+        final PipelineWithPermissions pipelineWithPermissions1 = new PipelineWithPermissions();
+        pipelineWithPermissions1.setId(pipeline1.getId());
+        pipelineWithPermissions1.setOwner(pipeline1.getOwner());
+        doReturn(pipelineWithPermissions1).when(mockPipelineWithPermissionsMapper).toPipelineWithPermissions(eq(pipeline1));
 
-        permissionsForPipeline1.put(TEST_OWNER1.toUpperCase(), 5);
-        permissionsForPipeline2.put(TEST_OWNER1.toUpperCase(), 1);
+        final PipelineWithPermissions pipelineWithPermissions2 = new PipelineWithPermissions();
+        pipelineWithPermissions2.setId(pipeline2.getId());
+        pipelineWithPermissions2.setOwner(pipeline2.getOwner());
+        doReturn(pipelineWithPermissions2).when(mockPipelineWithPermissionsMapper).toPipelineWithPermissions(eq(pipeline2));
+
+        doReturn(2).when(mockEntityManager).loadTotalCount(eq(AclClass.PIPELINE));
+        doReturn(2).when(mockEntityManager).loadTotalCount(AclClass.FOLDER);
+
+        permissionsForPipeline1.put(TEST_USER_1.toUpperCase(), 5);
+        permissionsForPipeline2.put(TEST_USER_1.toUpperCase(), 1);
         expectedMap.put(pipeline1.getId(), permissionsForPipeline1);
         expectedMap.put(pipeline2.getId(), permissionsForPipeline2);
+
+
         assertPipelineWithPermissions(expectedMap);
 
-        grantPermissionsToPipeline(AclPermission.WRITE.getMask(), testUserSid2.getSid(), pipeline1.getId());
-        permissionsForPipeline1.put(TEST_OWNER2.toUpperCase(), 4);
+    }
+
+    @Test
+    public void shouldReturnPermissionsWhenGrantedToPipelineCase2() {
+        // key - pipelineId, value - <sid, mask>
+        final Map<Long, Map<String, Integer>> expectedMap = new HashMap<>();
+        final Map<String, Integer> permissionsForPipeline1 = new HashMap<>();
+        final Map<String, Integer> permissionsForPipeline2 = new HashMap<>();
+
+        final List<Pair<AbstractSecuredEntity, List<AbstractGrantPermission>>> pipelinesAndFolders = Arrays.asList(
+                new ImmutablePair<>(pipeline1, Collections.singletonList(new UserPermission(TEST_USER_1, AclPermission.WRITE.getMask()))),
+                new ImmutablePair<>(pipeline1, Collections.singletonList(new UserPermission(TEST_USER_2, AclPermission.WRITE.getMask()))),
```

Comment:
I believe that you should specify permissions for a single entity using a single pair because new `initAclEntity` method just overrides already specified object entity permissions.

As a result in this test you give WRITE permissions for the first pipeline to the first user but in the `expectedMap` you specify that the first user should have READ permissions for the first pipeline.

```java
new ImmutablePair<>(pipeline1, Arrays.asList(
    new UserPermission(TEST_USER_1, AclPermission.WRITE.getMask()), 
    new UserPermission(TEST_USER_2, AclPermission.WRITE.getMask())))
```

Please check for similar cases in other tests too.